File: ProductService/src/Application/Interfaces/IRabbitMqService.js
class IRabbitMqService {
    async declareExchange(exchangeName, exchangeType) {
        throw new Error('Method not implemented');
    }

    async declareQueueAndBind(queueName, exchangeName, routingKeys) {
        throw new Error('Method not implemented');
    }

    async publish(exchangeName, routingKey, message) {
        throw new Error('Method not implemented');
    }

    async consume(queueName, onMessage) {
        throw new Error('Method not implemented');
    }

    async close() {
        throw new Error('Method not implemented');
    }
}

module.exports = IRabbitMqService;

File: NotificationService/src/Interfaces/IRabbitMqService.cs
namespace src.Interfaces
{
    public interface IRabbitMqService
    {
        void DeclareExchange(string exchangeName, string exchangeType);


        void DeclareQueueAndBind(string queueName, string exchangeName, string routingKeys);

        void Publish(string exchangeName, string routingKey, object message);

        void Subscribe(string queueName, Func<string, string, Task> onMessageAsync);
    }
}
